design/Proposal-to-enhance-schema-for-metal3.md

```diff
@@ -0,0 +1,228 @@
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+      * [References](#references)
+<!--te-->
+
+## Summary
+
+Enhancing the `metal3` schema by adding useful parameters that we received 
+from ironic introspection which will be helpful in performance based use cases.
+
+## Motivation
+
+To make the existing `metal3` schema compatible and scalable for heavy computation 
+based use cases.  
+
+### Goals
+
+To compare the introspection data fetched from `Ironic-Inspector` through 
+`Gopher-cloud client` library with `metal3` schema and add useful parameters 
+in `metal3` schema.
+
+## Proposal
+
+The comparison of hardware introspection data coming from `Ironic-Inspector` 
+and existing `metal3` schema for configuration of `baremetalhost` concludes 
+that few hardware parameters are missing in `metal3` schema that are fetched 
+from `Ironic-Inspector`. We propose to add these parameters in `metal3` schema for 
+scalability and performance enhancement.
+
+### Implementation Details/Notes/Constraints
+
+Link for Existing `Metal3` Specs
+
+Please refer `metal3` spec for bare-metal:
+https://github.com/metal3-io/baremetal-operator/blob/master/deploy/crds/metal3.io_baremetalhosts_crd.yaml
+
+The investigation of the existing schema indicates that BareMetal-Operator 
+gets BareMetal node introspection data using an API call to `Ironic-Inspector` 
+through `gopher-cloud client` library. 
+
+Having fetched BareMetal Node introspection data from Ironic-Inspector and
+compared it with BareMetalHost schema of `metal3` `(metal3.io_baremetalhosts_crd.yaml`), 
+we recommend following parameters need to be added in schema of `metal3` for `NFV` 
+workloads.
+1. Add enhancement in the existing `metal3` schema for the following parameters:
+
+    CPU Parameters:
+    
+    Parameters for `NUMA` topology:
+     
+    For better performance, selection of CPUs based on the `NUMA` topology 
+    becomes necessary. In case of nodes with `DPDK` aware NICs, the CPUs for 
+    poll mode driver (PMD) need to be selected from the `NUMA` node associated 
+    with the `DPDK` NICs. If hyperthreading is enabled, then selection of the 
+    logical cores requires the knowledge of the siblings.  
+
+    Declarative design idea:
+    ````yaml
+    Cpu
+     …
+    numa_topology:
+      properties:
+        numa_node_0:
+          properties:
+            nics:
+              properties:
+                numa_node:
+                  type: integer
+                  description: Numa node id
+                name:
+                  type: string
+                  description: network interface name
+              type: object
+            type: array
+            ram:
+              properties:
+                numa_node:
```

Comment:
Since we only track RAM as a total value elsewhere, it probably makes sense to keep this section to list the breakdown. But the same question applies as above: is the numa_node ID value here ever going to be different from the node ID from line 85?

Reply:
RAM value is same for any NUMA node (socket).

Reply:
ok.